libs/core/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,740 @@
+import { PopoverComponent } from '../popover/public_api';
+import { ControlValue, CustomValues, RangeHandles, SliderTickMark, SliderValueTargets } from './slider.model';
+import { MIN_DISTANCE_BETWEEN_TICKS } from './constants';
+
+export const SLIDER_VALUE_ACCESSOR = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => SliderComponent),
+    multi: true
+};
+
+let sliderId = 0;
+
+@Component({
+    selector: 'fd-slider',
+    templateUrl: './slider.component.html',
+    styleUrls: ['./slider.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [SLIDER_VALUE_ACCESSOR]
+})
+export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlValueAccessor, CssClassBuilder {
+    /** Slider id, it has some default value if not set,  */
+    @Input()
+    @HostBinding('attr.id')
+    id = 'fd-slider-id-' + sliderId++;
+
+    /** User's custom classes */
+    @Input()
+    class: string;
+
+    /** Id of the element that labels object number. */
+    @Input()
+    ariaLabelledBy: string = null;
+
+    /** Aria label for the object number. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Get Minimum value. */
+    @Input()
+    get min(): number {
+        return this._min;
+    }
+
+    /** Set Minimum value. */
+    set min(value: number) {
+        if (((this.max - value) / this.step) % 1 !== 0) {
+            return;
+        }
+
+        this._min = value;
+    }
+
+    /** Get Maximum value. */
+    @Input()
+    get max(): number {
+        return this._max;
+    }
+
+    /** Set Maximum value. */
+    set max(value: number) {
+        if (((value - this.min) / this.step) % 1 !== 0) {
+            return;
+        }
+
+        this._max = value;
+    }
+
+    /** Get Step value. */
+    @Input()
+    get step(): number {
+        return this._step;
+    }
+
+    /** Set Step value. */
+    set step(value: number) {
+        if (((this.max - this.min) / value) % 1 !== 0) {
+            return;
+        }
+
+        this._step = value;
+    }
+
+    /** Jump value. */
+    @Input()
+    jump = 10;
+
+    /** Put a label on every N-th tickmark. */
+    @Input()
+    tickmarksBetweenLabels = 1;
+
+    /**
+     * Slider mode.
+     * Options include: 'single' | 'range'
+     * The default is set to 'single'
+     */
+    @Input()
+    mode: 'single' | 'range' = 'single';
+
+    /** Toggles the visibility of tick marks. */
+    @Input()
+    showTicks = false;
+
+    /** Toggles the visibility of tick mark labels. Must be used in conjunction with 'showTicks' */
+    @Input()
+    showTicksLabels = false;
+
+    /** Array of custom values to use for Slider. */
+    @Input()
+    customValues: CustomValues[] = [];
+
+    /** Tooltip can be two types, 'readonly' to display value and 'editable' to make the ability to set and display value. */
+    @Input()
+    tooltipMode: 'readonly' | 'editable';
```

Comment:
Is  this component `readonly` by default?

Reply:
No, by default without tooltip